### PR TITLE
JBIDE-20175 JBIDE-20141 use correct URL for n...
...ightly/snapshots site; also fix requirements to state JDK 8 (not JDK 7 or 8)

### PR DESCRIPTION
**Pull request by nickboldt**

### CODE REVIEW

File: _config/products.yml [hunk 1]
```diff
@@ -592,7 +592,7 @@ jbt_core:
           - name: Release Notes
             url: http://tools.jboss.org/documentation/whatsnew/index.html
       4.3.0.Nightly: 
-        update_site_url: http://download.jboss.org/jbosstools/mars/development/nightly
+        update_site_url: http://download.jboss.org/jbosstools/updates/snapshots/mars/
```
Comment (maxandersen) [inline _config/products.yml]: the development/nightly seems like bad typo/copy/paste ;/

but why not http://download.jboss.org/jbosstools/mars/nightly/updates/ as we agreed on in chat for the release and used in past ?

Comment (nickboldt) [inline _config/products.yml]: /snapshots/ is the new /nightly/ 

See JBDS-2308.

**Comment by maxandersen** [comment]
this is not about snapshots vs nightly - its about using the uniform public urls vs pointing inside our internal urls.

JBDS-2308 is "Vet proposed Android skins with Legal"

**Comment by maxandersen** [comment]
As discussed in JBIDE-20141 there is still some mismatch between the urls so for now I've rolled this back to what we used the first 6 months: http://download.jboss.org/jbosstools/updates/nightly/mars/

once the repo location discussed at JBIDE-20141 has been settled so we don't move this again back and forth we can update.

**Pull request closed without merging**